tf/getter.go

```diff
@@ -380,3 +462,38 @@ func BuildRequestURL(registryDomain string, moduleRegistryBasePath string, modul
 
 	return &url.URL{Scheme: "https", Host: registryDomain, Path: moduleFullPath}, nil
 }
+
+// getHighestVersion returns the highest version from the list of versions, or an empty string if the list is empty.
+func getHighestVersion(availableVersions []*version.Version) string {
+	if len(availableVersions) == 0 {
+		return ""
+	}
+
+	sort.Sort(version.Collection(availableVersions))
+
+	return availableVersions[len(availableVersions)-1].String()
+}
+
+// getCompatibleVersions returns the list of versions that satisfy the version constraint.
+func getCompatibleVersions(availableVersions []ModuleVersion, versionQuery string) ([]*version.Version, error) {
+	// Filter the available versions based on the version constraint
+	filteredVersions := []*version.Version{}
```

Comment:
var filteredVersions []*version.Version
or even pre-allocate
filteredVersions := make([]*version.Version, 0, len(availableVersions))

Reply:
Since filteredVersions is a subset of availableVersions and the length can't be known in advance, I'll go for the first option. Thank you!